Rewrite domain restrictions

krve opened the pull request:
I've made a draft of how the domain validation could work according to #1

krve committed:
Change domain restriction-system

This will allow for a more configurable system where
you can specify a full domain name, all subdomains,
a tld and port numbers.

krve committed:
Update readme

krve committed:
Merge branch 'master' of github.com:krve/sudo-su

mrterryh commented:
This looks awesome, I'm going to have a proper look in the morning. Thanks so much for this @krve!

jedkirby commented:
Would be nice if we could add some tests into the domain restrictions to ensure it's working correctly

krve commented:
I'll see if I can't create some tests for the domain restrictions this weekend.

krve committed:
Add tests for DomainRestricter

I also extracted the domain restricter to a specific class
to keep it self-contained and make it easier for testing purposes

jedkirby reviewed (approved):
Nice work @krve, looks much better 👍

Jaspur commented:
But not really backwards compatible for people using current version?

mrterryh commented:
Sorry, been hectic at work today! Will definitely review this early next week.

krve commented:
Completely forgot about that @Jaspur. Will look into it this weekend to make sure it has some backwards capability.

mrterryh commented:
If you think backwards compatibility is going to be an issue, I would be happy to consider making this a breaking change and publishing a new version of the package.

krve commented:
Yeah, I think that's a better idea. This way we don't have code specifically for an old version of the package.

mrterryh commented:
Cool. I'm thinking about merging all these recent PRs and just releasing a new `2.0` version. Do you think that's sensible @krve?

krve commented:
@mrterryh yeah, that seems like a good idea.

mrterryh commented:
Just to let you know, I've still got this on my to-do list! Super hectic at work. I'll look ASAP, I promise!

DODMax commented:
I tried your package as I need custom domains but I get the following error:
```
Type error: Argument 1 passed to VIACreative\SudoSu\ServiceProvider::__construct() must be an instance of VIACreative\SudoSu\DomainRestricter, instance of Illuminate\Foundation\Application given, called in E:\xampp\htdocs\mfund\vendor\laravel\framework\src\Illuminate\Foundation\Application.php on line 612
```

krve commented:
@DODMax on what version of Laravel?

DODMax commented:
@krve on 5.4.25
I'm using commit 05fe05634f2aa503463b72dc5ad77dd09b50e259

krve commented:
@DODMax I'll take a look at it today when I'm home and fix the issue.

krve committed:
Change DomainRestricter initialization

Moved the initialization to the boot method and make use of laravels Dependency Injection.

krve commented:
@DODMax Just pushed an update that should fix the issue :+1:

DODMax commented:
@krve Works like a charm. Thanks for the PR!

krve commented:
@mrterryh - You wanna merge this? I kinda want to get this one merged so I can work on some of the other issues.

LoganGray commented:
I'm very surprised nothing ever happened with this.  I just hacked it a bit to work with an existing application that is still basically Laravel 5.2.  And while it had some issues.. I was able to get it to do what I wanted.  
I really like the concept.  Does anyone know if this has been improved and one of these many forks is actually preferred?

karmendra commented:
This error has been a pain, and i see this PR addresses it well, any ETA on when this is going to be released. I prefer not to use a fork as it will be a pending task to switch back to original repository, else would miss all the other enhancements.
